maze/README.rst

```diff
@@ -0,0 +1,78 @@
+Maze Solver Demo
+================
+Getting the D-Wave Quantum Computer to solve a maze!
+
+The purpose of this demo is to take a simple and familiar problem - solving a maze - and go through
+the steps of submitting the problem to the quantum computer.
+
+Code Overview
+-------------
+The idea is to describe the rules of how one can move through a maze with a set of constraints.
+Using the Ocean toolkit, these constraints can be converted into a binary quadratic model (BQM)
+that can then be solved with a D-Wave quantum computer. The solution that gets returned by the
+quantum computer is the path needed to get through the maze.
+
+There are several constraints involved with a maze:
+ - Specify valid path moves (ie if the path enters a grid point, it must also leave said grid point)
+ - Path has a specific start and end position
+ - Path cannot go beyond the border of the maze
+ - Path cannot go through the walls within the maze
+
+Each of these constraints are implemented by the :code:`maze` functions:
+:code:`_apply_valid_move_constraint()`, :code:`_set_start_and_end()`, :code:`_set_borders()`, and
+:code:`_set_inner_walls()`, respectively. These functions are called when the user calls
+:code:`get_maze_bqm(..)` in Example below.
+
+Code Specifics
+--------------
+The maze is a rectangular grid. The path segments (aka edges) that can be formed in this grid are
+described with respect to a grid point.
+
+Consider the edge labelled :code:`'1,0w'`:
+ - :code:`1,0` refers to grid point on row 1, column 0
+ - :code:`w` refers to "west"
+ - Hence, if you imagine a compass that is centered at position :code:`1,0`, the edge :code:`'1,0w'`
+   is the west "spoke" of this compass
+
+Note that the code only accepts edge inputs in the north direction (:code:`'<row>,<col>n'`) and the
+west direction (:code:`'<row>,<col>w'`). So if edges in the south or east directions are needed,
+please rewrite in terms of north and west. Namely,
```

Comment:
"So if edges in the south or east directions are needed" --> I's assume they will be needed so maybe "Edges in south or east directions can be restated as edges in north and west directions:"